api/storage_service.cc

```diff
@@ -1556,3 +1549,20 @@ void unset_snapshot(http_context& ctx, routes& r) {
 }
 
 }
+
+future<> major_keyspace_compaction_task_impl::run() {
```

Comment:
No one will think to look for the definition of something declared in `compaction/*.hh` in `api/*.cc`. Please create a `compaction/compaction_task.cc` and move these definitions there.